provider/machine_pool_resource.go

```diff
@@ -219,6 +251,46 @@ func (r *MachinePoolResource) Create(ctx context.Context,
 		return
 	}
 
+	isMultiAZPool, err := r.validateAZConfig(state)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Can't build machine pool",
+			fmt.Sprintf(
+				"Can't build machine pool for cluster '%s': %v",
+				state.Cluster.Value, err,
+			),
+		)
+		return
+	}
+	azList, err := common.StringListToArray(state.AvailabilityZones)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Can't build machine pool",
+			fmt.Sprintf(
+				"Can't build machine pool for cluster '%s': %v",
+				state.Cluster.Value, err,
+			),
+		)
+		return
+	}
+	if len(azList) > 0 {
+		builder.AvailabilityZones(azList...)
+	}
+	subnetList, err := common.StringListToArray(state.SubnetIDs)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Can't build machine pool",
+			fmt.Sprintf(
+				"Can't build machine pool for cluster '%s': %v",
+				state.Cluster.Value, err,
+			),
+		)
+		return
+	}
+	if len(subnetList) > 0 {
+		builder.Subnets(subnetList...)
+	}
+
```

Comment:
What if `len(subnetList) > 0 && len(azList) > 0` ? Would we like to block it?

Reply:
I decided not to because once it has been created, OCM populates the az list in the create response. If the user had supplied subnets, now we'd have both in the state file, causing the "both" test to fail when an update is applied (e.g., scaling).

Reply:
I understand the issue in update scenario, but this validation is for create scenario. we can have this validation only for create, not for update. am I right?